### PR TITLE
Grant @b-n write access to code repositories

### LINKED ISSUE #296: Grant @b-n commit rights
@b-n has made some amazing contributions to Artichoke:

- Adding automation around Rust toolchain upgrades: https://github.com/artichoke/artichoke/pull/1583
- Adding a UI test harness: https://github.com/artichoke/artichoke/pull/1584
- Executing a huge refactor for `String` which improved performance, maintainability, and extensibility: https://github.com/artichoke/artichoke/pull/1678
- Adding a second backend to `spinoso-time` using `tz-rs`: https://github.com/artichoke/artichoke/pull/1902

And many other changes.

@b-n's contributions are large, impactful, and demonstrate a mastery of Artichoke's codebase, design, and vision.

Congrats @b-n!

### PR DESCRIPTION
**Pull request by lopopolo**
Fixes #296.

@b-n is the first external contributor being added to the Artichoke GitHub organization.

This PR grants access to the following repositories:

- artichoke
- boba
- cactusref
- clang-format
- focaccia
- intaglio
- jasper
- playground
- qed
- rand_mt
- raw-parts
- roe
- ruby-file-expand-path
- strftime-ruby
- strudel

These changes create a new team called @artichoke/contributors to manage this access.

This team does NOT have access to any repos that manage web properties (with the exception of artichoke/playground), repos that generate releases (like nightly and docker-artichoke-nightly), artichoke/logo, or forks.

## terraform plan

```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following
symbols:
  + create

Terraform will perform the following actions:

  # github_team_repository.contributor_repos["artichoke"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "artichoke"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["boba"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "boba"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["cactusref"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "cactusref"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["clang-format"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "clang-format"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["focaccia"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "focaccia"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["intaglio"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "intaglio"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["jasper"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "jasper"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["playground"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "playground"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["qed"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "qed"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["rand_mt"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "rand_mt"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["raw-parts"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "raw-parts"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["roe"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "roe"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["ruby-file-expand-path"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "ruby-file-expand-path"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["strftime-ruby"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "strftime-ruby"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["strudel"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "strudel"
      + team_id    = (known after apply)
    }

  # module.org_members.github_membership.members["b-n"] will be created
  + resource "github_membership" "members" {
      + etag     = (known after apply)
      + id       = (known after apply)
      + role     = "member"
      + username = "b-n"
    }

  # module.team_contributors.github_team.this will be created
  + resource "github_team" "this" {
      + create_default_maintainer = false
      + description               = "Code contributors"
      + etag                      = (known after apply)
      + id                        = (known after apply)
      + members_count             = (known after apply)
      + name                      = "contributors"
      + node_id                   = (known after apply)
      + privacy                   = "secret"
      + slug                      = (known after apply)
    }

  # module.team_contributors.github_team_members.this will be created
  + resource "github_team_members" "this" {
      + etag    = (known after apply)
      + id      = (known after apply)
      + team_id = (known after apply)

      + members {
          + role     = "maintainer"
          + username = "lopopolo"
        }
      + members {
          + role     = "member"
          + username = "b-n"
        }
    }

Plan: 18 to add, 0 to change, 0 to destroy.
```

### CODE REVIEW

**Comment by lopopolo** [comment]
This change is not applied.

@b-n how does this sound to you?

**Comment by lopopolo** [comment]
We should probably also chat around what "commit access" means and how I may need to adjust my contribution methods going forward.

**Review by b-n — APPROVED** [review]
LGTM! (just wanted to press the approve button)

**Comment by b-n** [comment]
Some ideas from my current workplace with contribution:

- Administrators have the ability to override branch checks (e.g. can use administrator access to force push a branch etc)
- Contributors can create PRs, branches, tags, etc, but require at least 1 approval before merging to trunk (e.g. can't push to trunk without a review)
- After approval, the contributor is the one that presses the merge button, rather than the approver - but this is because not everything in our development process is CI automated (some small things require some contributor attention etc)
- In practise, administrators always go through review process too (somehow our security team gets a notification if someone pushes directly to master).
- With dependabot PRs, if it's simple changes (e.g. following the Change log of the dependency), then we usually just self assign/approve/merge. For larger and/or more structural changes (e.g. major versions), we generally ask for review from others etc.

This is just some ideas on how to model contributor access without dangerously changing master branches that have CI on them etc. IMO, I'd imagine my future contributions should still always be reviewed. I'm happy to review some of your PRs, but I would not say it's by any means mandatory or the "normal process" - you clearly have much more context and knowledge etc!

**Comment by lopopolo** [comment]
> * Administrators have the ability to override branch checks (e.g. can use administrator access to force push a branch etc)

I think this is currently set up. All repositories have branch protection rules for `trunk`: https://github.com/artichoke/project-infrastructure/issues/299.

> * Contributors can create PRs, branches, tags, etc, but require at least 1 approval before merging to trunk (e.g. can't push to trunk without a review)

This is not enforced,  but seems like a good process for us to try and follow.

> * After approval, the contributor is the one that presses the merge button, rather than the approver - but this is because not everything in our development process is CI automated (some small things require some contributor attention etc)

Sounds good to me.

> * In practise, administrators always go through review process too (somehow our security team gets a notification if someone pushes directly to master).
> * With dependabot PRs, if it's simple changes (e.g. following the Change log of the dependency), then we usually just self assign/approve/merge. For larger and/or more structural changes (e.g. major versions), we generally ask for review from others etc.

For dependabot, usually for Ruby, GitHub Actions changes I merge those directly. For Rust and npm, to limit churn and CI time, I make all updates in bulk in 1 PR.

**Comment by lopopolo** [comment]
This change is applied:

```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following
symbols:
  + create

Terraform will perform the following actions:

  # github_team_repository.contributor_repos["artichoke"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "artichoke"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["boba"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "boba"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["cactusref"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "cactusref"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["clang-format"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "clang-format"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["focaccia"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "focaccia"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["intaglio"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "intaglio"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["jasper"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "jasper"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["playground"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "playground"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["qed"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "qed"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["rand_mt"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "rand_mt"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["raw-parts"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "raw-parts"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["roe"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "roe"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["ruby-file-expand-path"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "ruby-file-expand-path"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["strftime-ruby"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "strftime-ruby"
      + team_id    = (known after apply)
    }

  # github_team_repository.contributor_repos["strudel"] will be created
  + resource "github_team_repository" "contributor_repos" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "push"
      + repository = "strudel"
      + team_id    = (known after apply)
    }

  # module.org_members.github_membership.members["b-n"] will be created
  + resource "github_membership" "members" {
      + etag     = (known after apply)
      + id       = (known after apply)
      + role     = "member"
      + username = "b-n"
    }

  # module.team_contributors.github_team.this will be created
  + resource "github_team" "this" {
      + create_default_maintainer = false
      + description               = "Code contributors"
      + etag                      = (known after apply)
      + id                        = (known after apply)
      + members_count             = (known after apply)
      + name                      = "contributors"
      + node_id                   = (known after apply)
      + privacy                   = "secret"
      + slug                      = (known after apply)
    }

  # module.team_contributors.github_team_members.this will be created
  + resource "github_team_members" "this" {
      + etag    = (known after apply)
      + id      = (known after apply)
      + team_id = (known after apply)

      + members {
          + role     = "maintainer"
          + username = "lopopolo"
        }
      + members {
          + role     = "member"
          + username = "b-n"
        }
    }

Plan: 18 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

module.org_members.github_membership.members["b-n"]: Creating...
module.team_contributors.github_team.this: Creating...
module.org_members.github_membership.members["b-n"]: Creation complete after 4s [id=artichoke:b-n]
module.team_contributors.github_team.this: Creation complete after 6s [id=6269781]
github_team_repository.contributor_repos["qed"]: Creating...
github_team_repository.contributor_repos["playground"]: Creating...
github_team_repository.contributor_repos["ruby-file-expand-path"]: Creating...
github_team_repository.contributor_repos["raw-parts"]: Creating...
github_team_repository.contributor_repos["jasper"]: Creating...
github_team_repository.contributor_repos["clang-format"]: Creating...
module.team_contributors.github_team_members.this: Creating...
github_team_repository.contributor_repos["strftime-ruby"]: Creating...
github_team_repository.contributor_repos["artichoke"]: Creating...
github_team_repository.contributor_repos["rand_mt"]: Creating...
github_team_repository.contributor_repos["playground"]: Still creating... [10s elapsed]
github_team_repository.contributor_repos["raw-parts"]: Still creating... [10s elapsed]
github_team_repository.contributor_repos["qed"]: Still creating... [10s elapsed]
github_team_repository.contributor_repos["ruby-file-expand-path"]: Still creating... [10s elapsed]
github_team_repository.contributor_repos["jasper"]: Still creating... [10s elapsed]
github_team_repository.contributor_repos["clang-format"]: Still creating... [10s elapsed]
github_team_repository.contributor_repos["strftime-ruby"]: Still creating... [10s elapsed]
module.team_contributors.github_team_members.this: Still creating... [10s elapsed]
github_team_repository.contributor_repos["artichoke"]: Still creating... [10s elapsed]
github_team_repository.contributor_repos["rand_mt"]: Still creating... [10s elapsed]
github_team_repository.contributor_repos["qed"]: Creation complete after 17s [id=6269781:qed]
github_team_repository.contributor_repos["strudel"]: Creating...
github_team_repository.contributor_repos["raw-parts"]: Creation complete after 17s [id=6269781:raw-parts]
github_team_repository.contributor_repos["intaglio"]: Creating...
github_team_repository.contributor_repos["jasper"]: Creation complete after 18s [id=6269781:jasper]
github_team_repository.contributor_repos["cactusref"]: Creating...
github_team_repository.contributor_repos["ruby-file-expand-path"]: Creation complete after 18s [id=6269781:ruby-file-expand-path]
github_team_repository.contributor_repos["focaccia"]: Creating...
github_team_repository.contributor_repos["playground"]: Creation complete after 18s [id=6269781:playground]
github_team_repository.contributor_repos["roe"]: Creating...
github_team_repository.contributor_repos["clang-format"]: Creation complete after 18s [id=6269781:clang-format]
github_team_repository.contributor_repos["boba"]: Creating...
github_team_repository.contributor_repos["strftime-ruby"]: Creation complete after 18s [id=6269781:strftime-ruby]
github_team_repository.contributor_repos["artichoke"]: Creation complete after 19s [id=6269781:artichoke]
github_team_repository.contributor_repos["rand_mt"]: Creation complete after 19s [id=6269781:rand_mt]
module.team_contributors.github_team_members.this: Creation complete after 20s [id=6269781]
github_team_repository.contributor_repos["strudel"]: Still creating... [10s elapsed]
github_team_repository.contributor_repos["intaglio"]: Still creating... [10s elapsed]
github_team_repository.contributor_repos["cactusref"]: Still creating... [10s elapsed]
github_team_repository.contributor_repos["focaccia"]: Still creating... [10s elapsed]
github_team_repository.contributor_repos["roe"]: Still creating... [10s elapsed]
github_team_repository.contributor_repos["boba"]: Still creating... [10s elapsed]
github_team_repository.contributor_repos["strudel"]: Creation complete after 11s [id=6269781:strudel]
github_team_repository.contributor_repos["intaglio"]: Creation complete after 12s [id=6269781:intaglio]
github_team_repository.contributor_repos["cactusref"]: Creation complete after 11s [id=6269781:cactusref]
github_team_repository.contributor_repos["focaccia"]: Creation complete after 11s [id=6269781:focaccia]
github_team_repository.contributor_repos["roe"]: Creation complete after 11s [id=6269781:roe]
github_team_repository.contributor_repos["boba"]: Creation complete after 11s [id=6269781:boba]

Apply complete! Resources: 18 added, 0 changed, 0 destroyed.
```

**Pull request merged**
